### ISSUE TITLE
Race condition in include_from_node1

### ISSUE THREAD

**Issue by simonster** [JuliaLang/julia#11093]
As I noted in https://github.com/JuliaLang/julia/issues/11092#issuecomment-98241099, `include_from_node1` is vulnerable to a race condition when Julia has more than one worker, if another task gets run while it's sleeping and starts loading a file. Not a big deal, but here's an example that triggers it:

``` julia
$ julia -p1

[...]

julia> function f()
           @async reload("HDF5")
           reload("Distributions")
       end;

julia> f()
ERROR: LoadError: LoadError: could not open file /Users/simon/.julia/v0.4/ArrayViews/src/JLD.jl
 in include at ./boot.jl:250
 in include_from_node1 at ./loading.jl:129
 in include at ./boot.jl:250
 in include_from_node1 at ./loading.jl:129
 in reload_path at ./loading.jl:153
 in _require at ./loading.jl:68
 in require at ./loading.jl:54
 in include at ./boot.jl:250
 in include_from_node1 at ./loading.jl:129
 in reload_path at ./loading.jl:153
 in reload at loading.jl:86
 in f at none:3
while loading /Users/simon/.julia/v0.4/ArrayViews/src/ArrayViews.jl, in expression starting on line 2
while loading /Users/simon/.julia/v0.4/Distributions/src/Distributions.jl, in expression starting on line 3
```

**Comment by tkelman** [comment]
was this fixed by #18504?

**Comment by StefanKarpinski** [comment]
This is just a bug and doesn't seem to be a new one at that.
